app/src/main/java/org/mozilla/fenix/onboarding/HomeOnboardingDialogFragment.kt

```diff
@@ -37,7 +45,61 @@ class HomeOnboardingDialogFragment : DialogFragment() {
             context?.settings()?.let { settings ->
                 settings.hasShownHomeOnboardingDialog = true
             }
+            showJumpCFR()
             dismiss()
         }
     }
+
+    private fun showJumpCFR() {
+        val jumpBackInView = findJumpBackInView()
+        jumpBackInView?.let {
+            val crfDialog = createJumpCRF(anchor = jumpBackInView)
+            crfDialog.show()
+        }
+    }
+
+    private fun findJumpBackInView(): View? {
+        val list = activity?.findViewById<RecyclerView>(R.id.sessionControlRecyclerView)
+        val count = list?.adapter?.itemCount ?: return null
+
+        for (index in 0..count) {
+            val viewHolder = list.findViewHolderForAdapterPosition(index)
+            if (viewHolder is RecentTabsHeaderViewHolder) {
+                return viewHolder.containerView
+            }
+        }
+        return null
+    }
+
+    private fun createJumpCRF(anchor: View): Dialog {
+        val context: Context = requireContext()
+        val anchorPosition = IntArray(2)
+        val popupBinding = OnboardingJumpBackInCfrBinding.inflate(LayoutInflater.from(context))
+        val popup = Dialog(context)
+
+        popup.apply {
+            setContentView(popupBinding.root)
+            setCancelable(false)
+            // removing title or setting it as an empty string does not prevent a11y services from assigning one
+            setTitle(" ")
+        }
+        popupBinding.closeInfoBanner.setOnClickListener {
+            popup.dismiss()
+        }
+
+        anchor.getLocationOnScreen(anchorPosition)
```

Comment:
TIL!

It looks like this method defaults to (0, 0) if it can't find a view. Would we need an early-return check in case we can't find this item? Maybe the RecyclerView adapter has the item, but it's not on the screen because it doesn't fit?

I genuinely do not know since I've never used this approach before, but just throwing this out there. 🙂

Reply:
Thanks for catching this, I added the early-return check  :)